cmd/kube-proxy/app/server.go

```diff
@@ -361,6 +364,7 @@ type ProxyServer struct {
 	Client                 clientset.Interface
```

Comment:
This struct probably needs to be decomposed into smaller pieces rather than the union of all proxy modes.